### DISCUSSION TITLE
Filtering with centralized stitching

### DISCUSSION THREAD

**Discussion by Arik-Shapiro** [ChilliCream/hotchocolate#4054] [Q&A]
Is it possible to use filtering when using centralized stitching? 
https://github.com/ChilliCream/hotchocolate-examples/tree/master/misc/Stitching/centralized
In this example how would you add product filtering to the Stitching.graphql?
```
extend type Query {
  me: User! @delegate(schema: "accounts", path: "user(id: 1)")
  topProducts(first: Int = 5): [Product] @delegate(schema: "products")
  products: [Product] @delegate(schema: "products")
}
```

**Comment by PascalSenn** [comment]
Yo can add filtering to the downstream service and stitch it. But at the moment you cannot "add" filtering on the stitching layer

**Reply by Arik-Shapiro** [reply]
Thank you for your answer!
Just to clarify, In the example there's this code
```
extend type Product {
  reviews: [Review] @delegate(schema: "reviews" path:"reviewsByProduct(upc: $fields:upc)")
}
```
that delegates it to the function
```
        public IEnumerable<Review> GetReviewsByProduct(
            [Service] ReviewRepository repository,
            int upc) =>
            repository.GetReviewsByAuthorId(upc);
```

that means that in-order to stitch the reviews to the product I need a separate filtering function on ReviewsService?
can't I just use the regular GetReviews function with filtering?
something like this:
```
        [UseFiltering]
        public IEnumerable<Review> GetReviews(
            [Service] ReviewRepository repository) =>
            repository.GetReviews();


extend type Product {
  reviews: [Review] @delegate(schema: "reviews" path:"reviews(where: {upc: {eq: $fields:upc}})")
}

```

Sorry if this is a silly question.

edit: 
it actually works with 
```
extend type Product {
  reviews: [Review] @delegate(schema: "reviews" path:"reviews(where: {upc: {eq: 1}})")
}
```

but "$fields:upc" returns an error
            "extensions": {
            "message": "Unexpected token: $.",
            ...

**Reply by vickytr44** [reply]
Is there a way to pass $fields:upc to the where clause?
